src/main/java/net/sf/jabref/gui/actions/CopyDoiUrlAction.java

```diff
@@ -0,0 +1,45 @@
+package net.sf.jabref.gui.actions;
+
+import java.awt.event.ActionEvent;
+import java.util.Optional;
+
+import javax.swing.AbstractAction;
+import javax.swing.text.JTextComponent;
+
+import net.sf.jabref.JabRefGUI;
+import net.sf.jabref.gui.ClipBoardManager;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.util.DOI;
+
+/**
+ * Copies the doi url to the clipboard
+ */
+public class CopyDoiUrlAction extends AbstractAction {
+
+    private JTextComponent component = null;
+    private String identifier;
+
+    public CopyDoiUrlAction(String identifier) {
+        super(Localization.menuTitle("Copy DOI url"));
+        this.identifier = identifier;
+    }
+
+    public CopyDoiUrlAction(JTextComponent component) {
+        this(component.getText());
+        this.component = component;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        identifier = component == null ? identifier : component.getText();
```

Comment:
Please split that in an if/else, because that ternary operator and the assignment in one statement make it hard to understand for non java exports, especially if you don't know about the operator precedence.